solr/core/src/java/org/apache/solr/cli/PostTool.java

```diff
@@ -815,7 +820,7 @@ public void postFile(File file, OutputStream output, String type) throws Malform
           urlStr =
               appendParam(urlStr, "literal.id=" + URLEncoder.encode(file.getAbsolutePath(), UTF_8));
         }
-        url = new URL(urlStr);
+        url = URI.create(urlStr).toURL();
```

Comment:
Curious if using a URI would be better than constantly converting back to string?   Maybe another pr...